processor/tailsamplingprocessor/config.go

```diff
@@ -40,8 +40,43 @@ const (
 	StringAttribute PolicyType = "string_attribute"
 	// RateLimiting allows all traces until the specified limits are satisfied.
 	RateLimiting PolicyType = "rate_limiting"
+	// Composite allows defining a composite policy, combining the other policies in one
+	Composite PolicyType = "composite"
 )
 
+// SubPolicyCfg holds the common configuration to all policies under composite policy.
+type SubPolicyCfg struct {
```

Comment:
@jpkrohling Can you define a common struct for them, embedding it into the PolicyCfg and here? This way, there's no risk of getting one updated and not the other.

@vikrambe 
We can use same struct and eliminate subPolicyCfg, but before that we need to fix field checks in configcheck in opentelemetry-collector codebase. It does not support composites at the moment . With take this refactoring in next PR
goroutine 1 [running]:
reflect.name.tag(0x3374265, 0x0, 0x0)
/usr/local/go/src/reflect/type.go:502 +0x85 fp=0xc020c00388 sp=0xc020c00380 pc=0x497745
reflect.(*structType).Field(0x38d6aa0, 0x0, 0x3374268, 0xb, 0x0, 0x0, 0x45d29c8, 0x3535de0, 0x0, 0x0, ...)
/usr/local/go/src/reflect/type.go:1197 +0x116 fp=0xc020c003e0 sp=0xc020c00388 pc=0x49a8f6
reflect.(*rtype).Field(0x38d6aa0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
/usr/local/go/src/reflect/type.go:923 +0x8c fp=0xc020c004d0 sp=0xc020c003e0 pc=0x4991cc
go.opentelemetry.io/collector/config/configcheck.validateConfigDataType(0x45d29c8, 0x38d6aa0, 0x4488cf6, 0x1)
/home/obelix/go/pkg/mod/go.opentelemetry.io/collector@v0.33.1-0.20210827152330-09258f969908/config/configcheck/configcheck.go:92 +0xfa fp=0xc020c00648 sp=0xc020c004d0 pc=0xb6603a
go.opentelemetry.io/collector/config/configcheck.checkStructFieldTags(0x3332187, 0xa, 0x0, 0x0, 0x45d29c8, 0x38d6aa0, 0x3332193, 0x16, 0x20, 0xc000b4a488, ...)